packages/dart_firebase_admin/lib/src/utils/crypto_signer.dart

```diff
@@ -107,11 +107,33 @@ class _ServiceAccountSigner implements CryptoSigner {
 
   @override
   Future<Uint8List> sign(Uint8List buffer) async {
-    final key = utf8.encode(credential.privateKey);
-    final hmac = Hmac(sha256, key);
-    final digest = hmac.convert(buffer);
+    final rsaPrivateKey = _parsePrivateKeyFromPem();
```

Comment:
Just a suggestion: https://github.com/jonasroussel/dart_jsonwebtoken comes with a convenient `RSAPrivateKey(String pem)` constructor. Could this be used instead of all the ASN1 code?

Reply:
This seems to do the trick (it passes the test in `crypto_signer_test.dart`):
`return JWTAlgorithm.RS256.sign(RSAPrivateKey(credential.privateKey), buffer);`